Club join

Jchuang7 opened the pull request.

Jchuang7 committed:
Announcement and Application Changes

Announcement routes set up for get all announcements, get announcement info, get club announcements, and create announcements. Create announcement button and modal set up for club page, changed formatting of club page to fit figma design mostly, still WIP. Application basic page set up for user, apply button shows up for application type clubs and redirects to an application page. Contents of that page still WIP. Setup for applications creation page is done.

Jchuang7 committed:
Delete announcement route and announcement list

Set up delete route for announcements based off announcement id, listed out announcements under announcement section in club page, formatting WIP.

Jchuang7 committed:
Revert "Delete announcement route and announcement list"

This reverts commit 4148253942d76e8a473453d3c97ee876474b2fd1.

Jchuang7 committed:
Delete route and formatting changews

Delete route set up for announcements, and set up announcement list in club page along with descriptions when clicked on. Also made styling changes to club page to fit figma formatting design.

Jchuang7 committed:
Application Routes Created

Created application routes

Jchuang7 committed:
Finished Application Creation Form and made route

Made a route for getting latest application from a clubId, and made the page for creating an application form.

Jchuang7 committed:
Form creation navigation and work on application form

Added redirect in club settings to application creation form and displayed club application questions to application form when applying.

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -141,33 +182,63 @@ export default function ClubPage() {
           isOwner={memberType === "owner"}
         />
       )}
+      {showCreateAnnModal && (
```
I'd just rename this (and other similar ones) `showCreateAnnouncementModal `, it's not egregiously long and it'd match the rest of the variables

varangill commented on server/src/services/announcementService.ts:
```diff
@@ -0,0 +1,47 @@
+    return announcement;
+}
+
+async function fetchAnnouncements() {
```
I don't think this endpoint will ever have an actual usecase so maybe we should just remove it?

varangill commented on server/src/routes/announcementRoutes.ts:
```diff
@@ -0,0 +1,11 @@
+import { Router } from "express";
```
Add API docs like in the other `Routes` files for each endpoint

varangill commented on server/src/database/migrations/20231203212528_initial-db.ts:
```diff
@@ -1,4 +1,4 @@
-import { Knex } from "knex";
+import { Knex, TableBuilder } from "knex";
```
Remove unused import

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -113,6 +141,19 @@ export default function ClubPage() {
+        </button>
+    );
+  };
+  console.log(user)
```
console.log is still here

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -11,25 +12,42 @@ export default function ClubPage() {
   useEffect(() => {
     getData(`users/membership?userId=${user?.id}&clubId=${id}`).then((res) => {
       setMemberType(res.membershipType);
+      console.log(res)
```
console.log is still here

varangill commented on client/src/components/AnnouncementCreationModal.tsx:
```diff
@@ -0,0 +1,82 @@
+            </div>
+        </Modal.Body>
+        <Modal.Footer>
+            {/* <Button variant="secondary"
```
Maybe uncomment this? It's not really necessary cause there's an X to close the modal anyways, but it'd be more consistent with other modals in the app

varangill commented on client/src/components/AnnouncementCreationModal.tsx:
```diff
@@ -0,0 +1,82 @@
+            announcementText,
+            announcementTime
+        })
+        alert("Announcement Created")
```
I'd avoid alerts, bad UX

varangill commented on client/src/components/ClubPage.tsx:
```diff
@@ -141,33 +182,63 @@ export default function ClubPage() {
+          <div class="scroll">
+            {announcements.map((announcement) => {
+              return (
+                <ul key={announcement["id"]} class="announcement" onClick={() => alert(announcement["announcementText"])}>
```
Same thing - I'd avoid alerts, a Modal would be quick to do and look much better

varangill commented on client/src/components/ApplicationFormCreation.tsx:
```diff
@@ -0,0 +1,112 @@
+        const newData = inputFields.filter(input => input.trim() !== '');
+        setListData([...listData, ...newData]);
+        createApplication();
+        alert("Submitted")
```
Another alert to remove

varangill commented on client/src/components/AnnouncementCreationModal.tsx:
```diff
@@ -0,0 +1,82 @@
+                <input type="text"
+                    placeholder="Title"
+                    value={announcementTitle}
+                    onChange={(e) => setTitle(e.target.value)}
```
I think the title should have a max character count- maybe like 80?

varangill commented on client/src/components/AnnouncementCreationModal.tsx:
```diff
@@ -0,0 +1,82 @@
+            </div>
+            <div>
+                <p class="text">Description</p>
+                <input type="text" 
```
When the description gets bigger than the text input box, it goes left to right which feels weird, and most actual announcements are gonna get bigger than the text input box since it's not very big. Could you make it expand downwards instead?

varangill commented on client/src/components/AnnouncementCreationModal.tsx:
```diff
@@ -0,0 +1,82 @@
+  const onSubmit = async () => {
+    const userId = user?.id;
+    const clubId = props.clubId
+    let announcementTime = new Date()
```
We can actually just have the [database automatically do timestamping for us](https://github.com/varangill/ClubHub/blob/master/server/src/database/migrations/20240220194148_clubs-timestamp.ts) which we should prob do here to be consistent with the other tables

varangill reviewed (changes requested):
Looks good overall, but a few comments on the code

Regarding the actual design, the announcements section doesn't match the one [we have designed on Figma](https://www.figma.com/file/8JuKuRHmTT0OMm6FyRmpWm/ClubHub). it's open to your discretion style-wise, but I really think that each announcement should display the title, author's name, date/time, and text on the actual club page without having to click in, and if the announcement text is too long, let the user click on it to open up a modal (not an alert)

I think it'd also be good to remove all the application stuff and move it off into its own branch/PR. If you think it'd take too much time then don't bother and just leave it as is, it's just kind of hard to review and follow the logic for it when it's incomplete and mixed in with the announcement stuff

I mentioned this in the files too, but any new/updated endpoints should get API docs made for them

Jchuang7 committed:
Resolved requested changes

Announcement Modal, wrapping text, adding description to announcement, removing alerts, removed unnecessary imports, and removed unneeded fetch call.

Jchuang7 committed:
Updated Swagger Docs

Updated Swagger Docs with the announcement routes

Jchuang7 committed:
Added automatic timestamping

Added automatic timestamping to announcement creation.

Jchuang7 committed:
Update announcementService.ts

Changed to retrieve data from newest to oldest.
